pkg/app/piped/driftdetector/cloudrun/detector.go

```diff
@@ -263,6 +263,8 @@ func (d *detector) loadHeadServiceManifest(app *model.Application, repo git.Repo
 			if err != nil {
 				return provider.ServiceManifest{}, fmt.Errorf("failed to copy the cloned git repository (%w)", err)
 			}
+			defer repo.Clean()
```

Comment:
@Warashi 
it seems that the `dir` is removed after executing the `loadHeadServiceManifest`
So we might not call `repo.Clean()`. WDYT?

```
		// We have to copy repository into another directory because
		// decrypting the sealed secrets or attaching files might change the git repository.
		if attachmentUsed || encryptionUsed {
			dir, err := os.MkdirTemp("", "detector-git-processing")
			if err != nil {
				return nil, fmt.Errorf("failed to prepare a temporary directory for git repository (%w)", err)
			}
			defer os.RemoveAll(dir)

			repo, err = repo.Copy(filepath.Join(dir, "repo"))
			if err != nil {
				return nil, fmt.Errorf("failed to copy the cloned git repository (%w)", err)
			}
			defer repo.Clean()

			repoDir = repo.GetPath()
			appDir = filepath.Join(repoDir, app.GitPath.Path)
		}
``` 
https://github.com/pipe-cd/pipecd/blob/c5216c91bffc906f57b30133f3d0e557f1b1888f/pkg/app/piped/driftdetector/kubernetes/detector.go#L270C1-L281C22

Reply:
@ffjlabo 

I think we should call `repo.Clean()` here because the 'repo.Clean()' does not only the directory deletion.
IMO, we should call some methods like `Clean()` or `Close()` everywhere because the caller doesn't know what happens with such methods. The caller only knows the cleanup methods exist.

Reply:
@Warashi 
> I think we should call repo.Clean() here because the 'repo.Clean()' not only does the directory deletion.

I got it. The `git worktree` needs additional operation other than deleting the directory to keep the consistency of the worktree as you said. 

> IMO, we should call some methods like Clean() or Close() everywhere because the caller doesn't know what happens with such methods. The caller only knows the cleanup methods exist.

Also I agree with it.

Reply:
A little off topic, but the detector reuses the one repo. `loadHeadServiceManifest` also uses it as an arg. So I feel a little bit confused about cleaning at this point.

But I noticed that this section is used temporarily and different from the actual repo passed as arg. 🙏